### ISSUE TITLE
Pager support

### ISSUE THREAD

**Issue by l4l** [lunaryorn/mdcat#50]
Would be nice to support system pager and redirect stdout to it. Probably by default it is nice to handle big output via that pager, if specified and works.

It should probably look like `Command::new(env::var("PAGER")?).stdin(Stdio::piped()).spawn()`

**Comment by lunaryorn** [comment]
Thank you for this suggestion, but I would not like mdcat to use a pager.  I don’t use less and would not like my tool have a default behaviour that doesn’t match my personal preferences.

I’m fine with separate executable, eg mdless, using a pager, but I won’t add it myself.  Pull requests welcome.

**Comment by emersion** [comment]
I had a script called `mdless`:

```shell
mdcat -c yes $1 | less -R
```

However that broke with a new `mdcat` release. Would you accept to add back a way to force colors?

**Comment by lunaryorn** [comment]
That’s the default behaviour now.  You don’t need that flag anymore.

**Comment by emersion** [comment]
Indeed, works -- thanks!

**Comment by amadeu01** [comment]
![image](https://user-images.githubusercontent.com/11631092/56855530-e4331880-691e-11e9-9049-abbb24ed605b.png)

I'm having problem using less.

I simply tried `mdcat file.md | less`

**Comment by gitbugr** [comment]
> I'm having problem using less.
> 
> I simply tried `mdcat file.md | less`

I was having the same issue using Mac OS. This works:
`mdcat file.md | less -r`

**Comment by friederbluemle** [comment]
> > I'm having problem using less.
> > I simply tried `mdcat file.md | less`
> 
> I was having the same issue using Mac OS. This works:
> `mdcat file.md | less -r`

@gitbugr - Thanks, this _almost_ works. This only minor problem are the "jump mark" indicators in iTerm2: First, they drawn in the incorrect positions, and the scrolling behavior is very odd. They will move up with the text when scrolling down (as expected), but stay in place when scrolling up (i.e. only the text will move down, the jump marks will remain at the same visible location). Is there anything that can be done to fix this?

**Comment by friederbluemle** [comment]
Just found that the following will print the iTerm2 jump marks in the correct location, but the scrolling problem still persists:

    mdcat REAMD.md | less -rX

**Comment by ericonr** [comment]
@friederbluemle I'm currently using the following function in fish shell:

```shell
function mdbat --argument file
	mdcat $file | bat -p
end
```

It's probably (definitely) overkill, but it guaranteed completely correct behavior for everything.

**Comment by lunaryorn** [comment]
Closed in favour of #152
